nixos/modules/services/web-apps/alps.nix

```
@@ -80,16 +97,33 @@ in {
       after = [ "network.target" "network-online.target" ];
 
       serviceConfig = {
-        ExecStart = ''
-          ${pkgs.alps}/bin/alps \
-            -addr ${cfg.bindIP}:${toString cfg.port} \
-            -theme ${cfg.theme} \
-            imaps://${cfg.imaps.host}:${toString cfg.imaps.port} \
-            smpts://${cfg.smtps.host}:${toString cfg.smtps.port}
-        '';
-        StateDirectory = "alps";
-        WorkingDirectory = "/var/lib/alps";
+        ExecStart = "${cfg.package}/bin/alps ${escapeShellArgs cfg.args}";
         DynamicUser = true;
+        ## This is desirable but would restrict bindIP to 127.0.0.1
+        #IPAddressAllow = "localhost";
+        #IPAddressDeny = "any";
```

We could make this conditional depending on a bind address option.

---

Then we'd have to parse `bindIP`. If it's okay to cover the most prominent cases `127.0.0.1` and `[::1]` then this would be easily doable, but in principle one would have to check whether it matches *any* loopback IP.

---

Hm, this might actually work, given that the chosen `bindIP` is not too exotic so that systemd can parse it (e.g. IPv6 zone identifiers are not supported I believe).
```suggestion
        IPAddressAllow = cfg.bindIP;
        IPAddressDeny = "any";
```

---

No, this settings is actually completely useless because it filters both ingress and egress and not just socket binding.
https://www.freedesktop.org/software/systemd/man/systemd.resource-control.html#IPAddressAllow=ADDRESS%5B/PREFIXLENGTH%5D%E2%80%A6